applications/openshift/worker/file_groupowner_kubelet_conf/rule.yml

```diff
@@ -6,22 +6,66 @@ platform: {{{ product }}}-node
 
 {{%- if product == "eks" %}}
 {{%- set kubeletconf_path = "/etc/kubernetes/kubelet/kubelet-config.json" %}}
+{{% set description %}}
+  Ensure that if the kubelet refers to a configuration file with the
+  <tt>--config</tt> argument, that file is owned by <tt>root:root</tt>.
+{{% endset %}}
+{{% set rationale %}}
+  The kubelet reads various parameters, including security settings, from a
+  config file specified by the <tt>--config</tt> argument. If this file is
+  specified you should restrict its file permissions to maintain the
+  integrity of the file. The file should be writable by only the
+  administrators on the system.
+{{% endset %}}
+{{% set ocil %}}
+  First, SSH to the relevant worker node:
+  To check to see if the Kubelet Service is running:
+  <tt>sudo systemctl status kubelet</tt>
+  The output should return <tt>Active: active (running) since..</tt>
+  Run the following command on each node to find the appropriate Kubelet
+  config file:
+  <tt>ps -ef | grep kubelet</tt>
+  The output of the above command should return something similar to
+  <tt>--config /etc/kubernetes/kubelet/kubelet-config.json</tt> which is the
+  location of the Kubelet config file.
+  Run the following command:
+  <tt>stat -c %U:%G /etc/kubernetes/kubelet/kubelet-config.json</tt>
+  The output of the above command is the Kubelet config file's ownership.
+  Verify that the ownership is set to <tt>root:root</tt>
+{{% endset %}}
 {{%- else %}}
 {{%- set kubeletconf_path = "/etc/kubernetes/kubelet.conf" %}}
+{{% set description %}}
+  Ensure that the <tt>kubelet.conf</tt> file ownership is set to
+  <tt>root:root</tt>.
+{{% endset %}}
+{{% set rationale %}}
+  The <tt>kubelet.conf</tt> file is the kubeconfig file for the node, and
```

Comment:
We have quite a few rules that check permissions and ownership, usually doubled for user and group checks. Wouldn't it be nicer to write macros in `shared/macros.jinja` that branch per distribution and expand to either ssh-ing to the nodes for EKS or generic Kubernetes distros and oc debug in a loop for OCP? That way, we could reduce the amount of boilerplate in the rules themselves.

Reply:
btw see both the definition and usage of `rhcos_node_login_instructions` for example.

Reply:
That's a good point. My initial goal here was to propose the content as written word-for-word in the various benchmarks.

You're suggesting we write macros that expand out different boilerplate content based on the product?

Reply:
Yeah. I didn't do any research on how easy or hard it would be, so it's just a suggestion. But I was thinking something like `cluster_node_login_instructions` that would provide the boilerplate and branch based on the distro, `ssh` for generic kube and `oc debug` for OCP..